pkg/goods/goods.go

```diff
@@ -30,19 +30,38 @@ func K0sBinarySHA256() (string, error) {
 //go:embed bins/*
 var binfs embed.FS
 
-// materializeBinaries materializes all binary files from inside bins directory.
+// materializeBinaries materializes all binary files from inside bins directory. If the
+// file already exists a copy of it is made first before overwriting it, this is done
+// because we can't overwrite a running binary. Copies are removed.
```

Comment:
why would a binary be running? I could see this being the case on reset+reinstall, but in that case we should fix our reset code

Reply:
in fact, "the file name already exists" should be a panic-worthy error IMO

Reply:
oh ok, that makes sense

do we restart local-artifact-mirror after rewriting that binary?

Reply:
No, as far as I know we can't do such a thing from within a container. Do you have any idea how we could achieve that ?

Reply:
I think I know how we can do that. Let me craft a different PR for that. I will ping back and we can discuss this there.